homeassistant/components/vasttrafik/config_flow.py

```diff
@@ -0,0 +1,384 @@
+        """Get the station ID from either station name or station ID.
+
+        This is used during the import process only.
+        """
+        if location.isdecimal():
+            station_data = StationData(
+                name=location, gid=location
+            )  # no good way to get name from GID
+        else:
+            api_result: StationData = planner.location_name(location)[0]
+            station_data = StationData(name=api_result["name"], gid=api_result["gid"])
+        return station_data
+
+    async def async_step_import(self, import_data: dict[str, Any]) -> ConfigFlowResult:
+        """Handle import from YAML configuration."""
+        self._async_abort_entries_match(
+            {CONF_KEY: import_data[CONF_KEY], CONF_SECRET: import_data[CONF_SECRET]}
+        )
+
+        if errors := await validate_api_credentials(self.hass, import_data):
+            return self.async_abort(reason=errors["base"])
+
+        planner: vasttrafik.JournyPlanner = await self.hass.async_add_executor_job(
+            vasttrafik.JournyPlanner, import_data[CONF_KEY], import_data[CONF_SECRET]
+        )
+
+        subentries = []
+        for departure in import_data.get(CONF_DEPARTURES, []):
+            station_data = self.get_station_data(planner, departure[CONF_FROM])
+            subentries.append(
+                ConfigSubentryData(
+                    title=departure.get(CONF_NAME, station_data["name"]),
+                    subentry_type="departure_board",
+                    data={
+                        CONF_STATION_GID: station_data["gid"],
+                        CONF_NAME: departure.get(CONF_NAME, station_data["name"]),
+                        CONF_HEADING: departure.get(CONF_HEADING),
+                        CONF_LINES: departure.get(CONF_LINES, []),
+                        CONF_TRACKS: departure.get(CONF_TRACKS, []),
+                        CONF_DELAY: departure.get(CONF_DELAY, DEFAULT_DELAY),
+                    },
+                    unique_id=None,
+                )
+            )
+
+        return self.async_create_entry(
+            title="Västtrafik",
+            data={
+                CONF_KEY: import_data[CONF_KEY],
+                CONF_SECRET: import_data[CONF_SECRET],
+            },
+            subentries=subentries,
+        )
+
+
+class VasttrafikSubentryFlow(ConfigSubentryFlow):
+    """Handle a subentry config flow for Västtrafik departure boards."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the subentry flow."""
+        super().__init__()
+        self._search_results: list[StationData] = []
+        self._selected_station: StationData | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> SubentryFlowResult:
+        """Handle the initial step of adding a departure board."""
+        if user_input is not None:
+            search_query = user_input.get("search_query", "").strip()
```

Comment:
```suggestion
            search_query = user_input["search_query"].strip()
```
Create a constant for `search_query` and use that